config/config.go

```diff
@@ -467,6 +475,9 @@ func New(validates ...Validate) (Config, error) {
 		}
 	}
 
+	// populdate chain ID
+	SetExternChainID(cfg.Chain.ExternChainID)
```

Comment:
if the way of reading this parameter is so strict, what about _plugins and etc.? Thus, I don't see the benefit.

Reply:
can you give more details? _plugins?
what I did is to make sure the external chain ID is loaded exactly one

Reply:
do you mean like "-plugin=gateway"?

this is doable but I think putting in config.yaml is an easier way for delegates.

Reply:
think about all the scenarios how this function will be called (we don't need this function, but set the _evmNetworkID in the New function), these complicated handlings are redundant.

Reply:
again my purpose is to make sure the external chain ID is loaded exactly one, i think it's better/safer to do that

Reply:
in what case this struct or function call could be thread unsafe? I don't see it